### ISSUE TITLE
Incorrect sorting of decimals values

### ISSUE THREAD

**Issue by BartGysens** [maxwroc/battery-state-card#625]
**Describe the bug**
Sorting by value sets decimal values as first - screenshot

**How to reproduce**
Basic

**Expected behavior**
Correct order of values

**YAML configuration**
```yaml

type: vertical-stack
cards:
  - type: horizontal-stack
    cards:
      - type: custom:mushroom-title-card
        title: CR2032 - 3V
        subtitle: 'Stock: 11'
      - type: picture
        image: /local/media/pile-cr2032.png
        hold_action:
          action: none
        tap_action:
          action: none
        style: |
          ha-card {
              width: 100px;
              border: none;
              background: none;
              box-shadow: none;
              padding-top: 20px;
          }
  - type: custom:battery-state-card
    secondary_info: '{last_changed}'
    sort: state
    colors:
      steps:
        - '#ff0000'
        - '#ffff00'
        - '#00ff00'
      gradient: true
    bulk_rename:
      - from: Battery Health
      - from: Batterij
    state_map:
      - from: 'off'
        to: 100%
      - from: unavailable
        to: 0%
    entities:
      - entity: sensor.ouverture_sensor_1_kamer_lavande_badkamer_battery
      - entity: sensor.ouverture_sensor_2_salon_venster_hifi_battery
      - entity: sensor.ouverture_sensor_3_kamer_camomille_venster_battery
      - entity: sensor.ouverture_sensor_4_kamer_aneth_venster_battery
      - entity: sensor.ouverture_sensor_5_kamer_calendula_venster_battery
      - entity: sensor.ouverture_sensor_6_kamer_lavande_slaapkamer_battery
      - entity: sensor.ouverture_sensor_7_keuken_venster_achteraan_batterij
      - entity: sensor.ouverture_sensor_8_inkomhal_voordeur_battery
      - entity: sensor.ouverture_sensor_9_sanitaire_achterdeur_battery
      - entity: sensor.ouverture_sensor_10_sanitaire_venster_rechts_battery
      - entity: sensor.ouverture_sensor_11_postbus_battery
      - entity: sensor.ouverture_sensor_12_sanitaire_diepvriezer_battery

```

**Screenshots**
![Overzicht-–-Home-Assistant](https://github.com/maxwroc/battery-state-card/assets/8970161/b911e25a-8676-456e-a8e5-a041c397ae4b)

**Version**
Version 3.1.0

**Comment by scodari** [comment]
Same issue for me.
I've set
`round: 0`
to get rid of the decimal values as a workaround until the bug will be fixed.

**Comment by maxwroc** [comment]
Can you please try to use at the card level config:
`default_state_formatting: false`

Please let me know whether this solves the issue

**Comment by scodari** [comment]
> Can you please try to use at the card level config: `default_state_formatting: false`
> 
> Please let me know whether this solves/mitigates the issue

That solves the issue for me

**Comment by maxwroc** [comment]
Ok thanks for the confirmation. This narrows down the possible root cause a lot. I know where is the problem now, I think

**Comment by scodari** [comment]
Let me know if I can help you with some more tests

**Comment by maxwroc** [comment]
Fixed in v3.1.1
